packages/percy-storybook/src/getStories.js

```diff
@@ -1,111 +1,70 @@
-      function removeEventListener() {}
-      function postMessage() {}
-      function terminate() {}
-
-      api.postMessage = postMessage;
-      api.addEventListener = addEventListener;
-      api.removeEventListener = removeEventListener;
-      api.terminate = terminate;
+// The function below needs to be in a template string to prevent babel from transforming it.
+// If babel transformed it, puppeteer wouldn't be able to evaluate it properly.
+// See: https://github.com/GoogleChrome/puppeteer/issues/1665#issuecomment-354241717
+// Also see comment below about Debugging page.evaluate
+const fetchStoriesFromWindow = `(async () => {
+  return await new Promise((resolve, reject) => {
+    const storiesKey = '${storiesKey}';
+    // Check if the window has stories every 100ms for up to 10 seconds.
+    // This allows 10 seconds for any async pre-tasks (like fetch) to complete.
+    // Usually stories will be found on the first loop.
+    var checkStories = function(timesCalled) {
+      if (window[storiesKey]) {
+        // Found the stories, return them.
+        resolve(window[storiesKey]);
+      } else if (timesCalled < 100) {
+        // Stories not found yet, try again 100ms from now
+        setTimeout(() => {
+          checkStories(timesCalled + 1);
+        }, 100);
+      } else {
+        reject(new Error(
+          'Stories not found on window within 10 seconds. ' +
+          "Check your call to serializeStories in your Storybook's config.js."
+        ));
+      }
+    };
+    checkStories(0);
+  });
+})()`;
 
-      return api;
-    }
-    window.Worker = MockWorker;
-`;
+export default async function getStories(options = {}) {
+  let launchArgs = [];
 
-// jsdom doesn't support localStorage yet.
-// We use localStorageMock to allow the user's preview.js to interact with localStorage.
-const localStorageMock = `
-    var localStorageMock = (function() {
-      var store = {};
-      return {
-        getItem: function(key) {
-          return store[key];
-        },
-        setItem: function(key, value) {
-          store[key] = value.toString();
-        },
-        clear: function() {
-          store = {};
-        }
-      };
-    })();
-    Object.defineProperty(window, 'localStorage', { value: localStorageMock });
-    `;
+  // Travis CI requires Chrome to be launched without the sandbox
+  // See https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#running-puppeteer-on-travis-ci
+  // See https://docs.travis-ci.com/user/chrome#Sandboxing
+  const isTravisCI = process.env.TRAVIS;
+  if (isTravisCI) {
+    launchArgs.push('--no-sandbox');
+  }
 
-// jsdom doesn't support matchMedia yet.
-const matchMediaMock = `
-    window.matchMedia = window.matchMedia || (() => {
-      return {
-        matches: false,
-        addListener: () => {},
-        removeListener: () => {},
-      };
-    });
-    `;
+  const browser = await puppeteer.launch({ headless: true, args: launchArgs });
```

Comment:
This will now require the user to have some sort of framebuffer, right? E.g. before it wouldn't need anything, now it needs some sort of `DISPLAY=...` and possibly Xvfb or such.

Reply:
@joscha puppeteer will use Chromium (or Chrome) Headless by default, so no framebuffer is required.